File: src/main/java/org/adoptopenjdk/jheappo/objects/RootUnknown.java
package org.adoptopenjdk.jheappo.objects;

/*
 * Copyright (c) 2018 Kirk Pepperdine.
 * Licensed under https://github.com/AdoptOpenJDK/jheappo/blob/master/LICENSE
 * Instructions: https://github.com/AdoptOpenJDK/jheappo/wiki
 */

import org.adoptopenjdk.jheappo.heapdump.EncodedChunk;

public class RootUnknown extends HeapObject {

    public final static int TAG = 0xFF;

    public RootUnknown(EncodedChunk buffer) {
        super(buffer);
    }

    @Override
    public String toString() {
        return "Root Unknown : " + getId();
    }
}

File: src/main/java/org/adoptopenjdk/jheappo/objects/RootUnknown.kt
package org.adoptopenjdk.jheappo.objects

/*
 * Copyright (c) 2018 Kirk Pepperdine.
 * Licensed under https://github.com/AdoptOpenJDK/jheappo/blob/master/LICENSE
 * Instructions: https://github.com/AdoptOpenJDK/jheappo/wiki
 */

import org.adoptopenjdk.jheappo.heapdump.EncodedChunk

class RootUnknown(buffer: EncodedChunk) : HeapObject(buffer) {

    companion object {
        const val TAG: UByte = 0xFFU
    }

    override fun toString(): String {
        return "Root Unknown : $id"
    }
}
